### PR TITLE
Rename record equality parameters to 'left' and 'right'

### LINKED ISSUE #49300: Record operator parameter names are 'r1' and 'r2' rather than 'left' and 'right'
image

According to https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-parameters#naming-operator-overload-parameters, the standard names are `left` and `right`. Roslyn IDE fixes currently generate operators with these names as well.

### PR DESCRIPTION
**Pull request by C-xC-c**
Fixes #49300

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=51973) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=51973) before we can accept your contribution.<br/><br/>:x: C-xC-c [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=51973)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=51973) it.</sub>

**Comment by jcouv** [comment]
In general, changing parameter names would be a breaking change since callers may use explicit argument names. But I think we're okay because those are operators.

We'd need a corresponding update to the spec:
https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/records.md#equality-members

**Comment by C-xC-c** [comment]
I have created a pull request for the required spec changes here https://github.com/dotnet/csharplang/pull/4555

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityOperatorBase.cs [hunk 1]
```diff
@@ -58,10 +58,10 @@ protected sealed override (TypeWithAnnotations ReturnType, ImmutableArray<Parame
                     Parameters: ImmutableArray.Create<ParameterSymbol>(
                                     new SourceSimpleParameterSymbol(owner: this,
                                                                     TypeWithAnnotations.Create(ContainingType, NullableAnnotation.Annotated),
-                                                                    ordinal: 0, RefKind.None, "r1", isDiscard: false, Locations),
+                                                                    ordinal: 0, RefKind.None, "left", isDiscard: false, Locations),
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityOperatorBase.cs:61]: >"left" [](start = 94, length = 6)

Please add a test to `RecordTests` that demonstrates the change. Perhaps:
```csharp
var source = "record R;";
var comp = CreateCompilation(source);
Assert.Equal("System.Boolean R.op_Equality(R? left, R? right)", comp.GetMember("R.op_Equality").ToTestDisplayString());
Assert.Equal("System.Boolean R.op_Inequality(R? left, R? right)", comp.GetMember("R.op_Inequality").ToTestDisplayString());
```

Comment (cston) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityOperatorBase.cs:61]: Never mind, it looks like there are a number of existing tests that should be affected.

---
In reply to: [597302570](https://github.com/dotnet/roslyn/pull/51973#discussion_r597302570) [](ancestors = 597302570)

Comment (C-xC-c) [reply src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityOperatorBase.cs:61]: I think I have fixed all of the affected tests.

**Review by cston — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Comment by RikkiGibson** [comment]
Planning to merge this afternoon, once we're sure that there's no objection from the language design team.

**Comment by RikkiGibson** [comment]
Thanks for your contribution @C-xC-c!

**Pull request merged**